File: src/bcrypt.ts
import * as bcrypt from 'packages/bcrypt@v0.4.1/mod.ts'

const rounds = 10

export const hashPassword = async (password: string): Promise<string> => {
  const salt = await bcrypt.genSalt(rounds)

  const hashedPassword = await bcrypt.hash(password, salt)

  return hashedPassword
}

export const comparePassword = (
  password: string,
  hashedPassword: string,
): Promise<boolean> => {
  return bcrypt.compare(password, hashedPassword)
}

File: app/bcrypt.py
import bcrypt

salt = bcrypt.gensalt()


def verify_password(plain_password: str, hashed_password: str):
    enconde_password = plain_password.encode("utf-8")
    return bcrypt.checkpw(enconde_password, hashed_password.encode("utf-8"))


def get_password_hash(password: str):
    enconde_password = password.encode()
    passowrd_hashed = bcrypt.hashpw(enconde_password, salt)
    return passowrd_hashed
